pkg/build/builder/common.go

```diff
@@ -84,15 +98,9 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 			},
 		},
 	}
-
+	// NOTE: not sure that this is needed anymore
 	// Reset ResourceVersion to avoid a conflict with other updates to the build
 	build.ResourceVersion = ""
```

Comment:
Is there a reason we need this when we update the revision but not when we update the status? Seems like this either belongs with the `c.UpdateDetails(build)` call (so we always clear `ResourceVersion` before updating) or both here and also in `updateBuildFailureReason` (so we clear `ResourceVersion` after either function modifies the build).